blog/2025-11-24-sharded-repodata-improvements.mdx

```diff
@@ -0,0 +1,265 @@
+and wanted to bring this same fast experience to users of conda. So, earlier this year we began development
+of our own sharded repodata implementation in Python for the [conda-libmamba-solver](https://github.com/conda/conda-libmamba-solver).
+
+In the next section, we share the performance improvements we've seen so far.
+
+## Performance improvements
+
+To compare the performance improvements between the sharded and traditional approach, we used two different
+environment creation scenarios: **python** and **data science**. The python scenario just fetches the package
+"python" and the data science scenario fetches the packages, "pandas", "plotly" and "scipy". We benchmarked
+just the repodata fetching itself (check out the script we used
+[here](https://github.com/travishathaway/perfpy-conda/blob/d749765e1c124fb518784a7a2d91f3748bd32c8c/scripts/sharded_repodata_fetching.py)).
+
+
+:::info
+We used a special, experimental **conda-forge-sharded** channel because at the time of
+profiling, sharded repodata was not available on conda-forge.
+
+The benchmarks were also run inside a `linux/arm64` Docker container running on an Apple M1 Pro.
+:::
+
+Furthermore, we also ran our comparison using a **cold** cache where nothing was present in conda's
+cache and everything was fetched via network requests and a **warm** cache where very few if any
+network requests were necessary.
+
+To get a complete picture of how these changes affected conda, we not only measured **total time**
+but total **network bandwidth** usage and **maximum memory usage**.
+
+### Total time
+
+Below are the comparisons between traditional and sharded fetching measuring **total time**.
+
+#### Total time with cold cache
+
+<Plot
+  data={[
+    {
+      x: ['Python', 'Data science'],
+      y: [21.471660, 22.360671],
+      type: 'bar',
+      name: 'Traditional',
+      marker: { color: '#e74c3c' },
+    },
+    {
+      x: ['Python', 'Data science'],
+      y: [2.012411, 2.844390],
+      type: 'bar',
+      name: 'Sharded',
+      marker: { color: '#2ecc71' },
+    },
+  ]}
+  layout={{
+    title: 'Total time with cold cache',
+    xaxis: { title: 'Package Scenario' },
+    yaxis: { title: 'Time (seconds)' },
+    barmode: 'group',
+    height: 500,
+  }}
+/>
+
+#### Total time with warm cache
+
+<Plot
+  data={[
+    {
+      x: ['Python', 'Data science'],
+      y: [10.246788, 11.268202],
+      type: 'bar',
+      name: 'Traditional',
+      marker: { color: '#e74c3c' },
+    },
+    {
+      x: ['Python', 'Data science'],
+      y: [0.670952, 1.012970],
+      type: 'bar',
+      name: 'Sharded',
+      marker: { color: '#2ecc71' },
+    },
+  ]}
+  layout={{
+    title: 'Total time with warm cache',
+    xaxis: { title: 'Package Scenario' },
+    yaxis: { title: 'Time (seconds)' },
+    barmode: 'group',
+    height: 500,
+  }}
+/>
+
+#### Key takeaways
+
+- Under both scenarios, we see 10x speed-up in fetching and parsing of repodata
```

Comment:
```suggestion
- Under both scenarios, we see a 10x speed-up in fetching and parsing of repodata
```

Reply:
My math on the first graph of the draft blog post says "10.6x" and "8x" improvements for the "Python" and "data science" cases.